### ISSUE TITLE
Investigate potential gap in bwc testing for search requests

### ISSUE THREAD

**Issue by cbuescher** [elastic/elasticsearch#78472]
While working on #75745 I accidentally broke the wire serialization of SearchRequest on 7.x and was suprised to find that not all hell broke lose when e.g. running the mixed cluster rest tests. I created an experimental PR that adds a new boolean flag to the SearchRequest serialization on 8.0 https://github.com/elastic/elasticsearch/pull/77889, which without a bwc layer should break communication with older 7.x nodes. It appears the only thing that failed was a CCS rolling upgrade test (SearchStatesIT) which I find to be a poot indicator of such a big break.
I'd like to understand better why the mixed cluster tests don't seem to be too sensitive to this kind of change and potentially modify them or add additional test that would surface these situations better.

**Comment by cbuescher** [comment]
It seems like we seldom need to serialize/deserialize SearchRequest. Changing serialization in e.g. ShardSearchRequest slightly breaks tons of tests in `:qa:mixed-cluster` as expected, so this might not be such a huge gap but something specific to SearchRequest. Perhaps its really only CCS scenarios in which we send SearchRequest over the wire.

**Comment by javanna** [comment]
Good point, I created #84481 to expand CCS tests coverage. The two may be related, as I think from your conclusion above CCS is one of the few cases where we serialize the `SearchRequest` (when `ccs_minimize_roundtrips` is enabled). I wonder if we should also have some more specific serialization unit tests for the SearchRequest? Would that have caught the wire serialization issue that was introduced?
